data_steward/cdr_cleaner/cleaning_rules/null_invalid_foreign_keys.py

```diff
@@ -21,84 +29,173 @@
+        :param dataset_id: Dataset where cleaning rules are to be applied
+        :param sandbox_dataset_id: Identifies the sandbox dataset to store rows
+        :return: a list of queries
+        """
+        queries_list = []
+        sandbox_queries_list = []
+
+        for table in self.affected_tables:
+            field_names = [
+                field['name'] for field in resources.fields_for(table)
+            ]
+            foreign_keys_flags = []
+            fields_to_join = []
+
+            for field_name in field_names:
+                if field_name in FOREIGN_KEYS_FIELDS and field_name != table + '_id':
+                    fields_to_join.append(field_name)
+                    foreign_keys_flags.append(field_name)
+
+            if fields_to_join:
+                col_exprs = []
+                for field in field_names:
+                    if field in fields_to_join:
+                        if field in foreign_keys_flags:
+                            col_expr = '{x}.'.format(x=field[:3]) + field
                     else:
-                        table_alias = _mapping_table_for(
-                            '{x}'.format(x=key)[:-3])
-                    join_expression.append(
-                        LEFT_JOIN.format(dataset_id=dataset_id,
-                                         prefix=key[:3],
-                                         field=key,
-                                         table=table_alias))
-
-            full_join_expression = " ".join(join_expression)
-            query = dict()
-            query[cdr_consts.QUERY] = INVALID_FOREIGN_KEY_QUERY.format(
-                cols=cols,
-                table_name=table,
-                dataset_id=dataset_id,
-                project=project_id,
-                join_expr=full_join_expression)
-            query[cdr_consts.DESTINATION_TABLE] = table
-            query[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
-            query[cdr_consts.DESTINATION_DATASET] = dataset_id
-            queries_list.append(query)
-    return queries_list
+                        col_expr = field
+                    col_exprs.append(col_expr)
+                cols = ', '.join(col_exprs)
+
+                join_expression = []
+                for key in FOREIGN_KEYS_FIELDS:
+                    if key in foreign_keys_flags:
+                        if key == 'person_id':
+                            table_alias = cdr_consts.PERSON_TABLE_NAME
+                        else:
+                            table_alias = self.get_mapping_tables(
+                                '{x}'.format(x=key)[:-3])
+                        join_expression.append(
+                            LEFT_JOIN.render(dataset_id=self.dataset_id,
+                                             prefix=key[:3],
+                                             field=key,
+                                             table=table_alias))
+
+                full_join_expression = " ".join(join_expression)
```

Comment:
Just continue with the previous suggestion, you could potentially do this here if those class methods are defined separately. 

```suggestion
            if self.has_foreign_key():
                cols = self.get_col_expression(table)
                join_expression = self.get_join_expression(table)
```